impeller/aiks/canvas_benchmarks.cc

```diff
@@ -0,0 +1,60 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/benchmarking/benchmarking.h"
+
+#include "impeller/aiks/canvas.h"
+
+namespace impeller {
+
+namespace {
+size_t DrawRect() {
+  Canvas canvas;
+  for (auto i = 0; i < 500; i++) {
+    canvas.DrawRect(Rect::MakeLTRB(0, 0, 100, 100),
+                    {.color = Color::DarkKhaki()});
+  }
+  return 500;
+}
+
+size_t DrawCircle() {
+  Canvas canvas;
+  for (auto i = 0; i < 500; i++) {
+    canvas.DrawCircle({100, 100}, 5, {.color = Color::DarkKhaki()});
+  }
+  return 500;
+}
+
+size_t DrawLine() {
+  Canvas canvas;
+  for (auto i = 0; i < 500; i++) {
+    canvas.DrawLine({0, 0}, {100, 100}, {.color = Color::DarkKhaki()});
+  }
+  return 500;
+}
+}  // namespace
+
+// A set of benchmarks that measures the CPU cost of encoding canvas operations.
+// These benchmarks do not measure the cost of conversion through the HAL, no
+// do they measure the GPU side cost of executing the required shader programs.
+template <class... Args>
+static void BM_CanvasRecord(benchmark::State& state, Args&&... args) {
+  auto args_tuple = std::make_tuple(std::move(args)...);
+  auto test_proc = std::get<decltype(&DrawRect)>(args_tuple);
```

Comment:
nit: can you name the type `decltype(&DrawRect)` instead of assuming that the passed in functions will match the DrawRect signature?

Reply:
Done